nucypher/characters/lawful.py

```diff
@@ -996,6 +835,153 @@ def retrieve():
         return controller
 
 
+class RetrievalClient:
+    """
+    Capsule frag retrieval machinery shared between Bob and Porter.
+    """
+
+    def __init__(self, learner: Learner):
+        self._learner = learner
+        self.log = Logger(self.__class__.__name__)
+
+    def _reencrypt(self,
+                   ursula: 'Ursula',
+                   reencryption_request: 'ReencryptionRequest',
+                   delegating_key: PublicKey,
+                   receiving_key: PublicKey,
+                   ) -> Dict['Capsule', 'VerifiedCapsuleFrag']:
+        """
+        Sends a reencryption request to a single Ursula and processes the results.
+
+        Returns reencrypted capsule frags matched to corresponding capsules.
+        """
+
+        middleware = self._learner.network_middleware
+
+        try:
+            response = middleware.reencrypt(ursula, bytes(reencryption_request))
+        except NodeSeemsToBeDown as e:
+            # TODO: What to do here?  Ursula isn't supposed to be down.  NRN
+            message = (f"Ursula ({ursula}) seems to be down "
+                       f"while trying to complete ReencryptionRequest: {reencryption_request}")
+            self.log.info(message)
+            raise RuntimeError(message) from e
```

Comment:
Any reason we don't want to just reraise the original exception instead of using RuntimeException?